include/aspect/material_model/interface.h

```diff
@@ -245,6 +245,39 @@ namespace aspect
       const typename DoFHandler<dim>::active_cell_iterator *cell;
     };
 
+    template <int dim>
+    struct MaterialModelDerivatives
+	{
+    	/**
+    	 * Constructor. Initialize the various arrays of this structure with the
+         * given number of quadrature points and (finite element) components.
+    	 * 
+         * @param n_points The number of quadrature points for which input
+         * quantities will be provided.
+         * @param n_comp The number of vector quantities (in the order in which
+         * the Introspection class reports them) for which input will be
+         * provided.
+    	 */
+    	MaterialModelDerivatives (const unsigned int n_points,
+    			                  const unsigned int n_comp);
+    	
+    	/**
+    	 * The derivatives of the viscosities
+    	 */
+        std::vector<double> dviscosities_du;
+    	std::vector<double> dviscosities_dp;
+    	std::vector<SymmetricTensor<2,dim> > dviscosities_dstrain_rate;
+		
+	/**
+	 * The derivatives of the strain rates
+	 */
+    	std::vector<SymmetricTensor<2,dim> > dstrain_rate_du;
```

Comment:
You won't need this variable. In fact, you can't compute it because the strain rate is not a function of the velocity but of the derivatives of the velocity instead. But I'll show you tomorrow why you won't need it either.